examples/plugins/inmemkv/inmemkv_backend.cpp

```
@@ -0,0 +1,455 @@
+/**
+ * @brief Construct INMEMKV backend engine
+ *
+ * Step 1: Read localAgent from init_params.
+ * Step 2: In-memory store only (no filesystem).
+ * Logs: INMEMKV backend initialized ; Local agent =
+ */
+nixlInMemKVEngine::nixlInMemKVEngine(const nixlBackendInitParams *init_params)
+    : nixlBackendEngine(init_params) {
+
+    // 1. Local agent name
+    localAgent = init_params->localAgent;
```

Cppcheck (2.20.0)

[performance] Variable 'localAgent' is assigned in constructor body. Consider performing initialization in initialization list.

(useInitializationList)